examples/analysis/ubet/eval_nitro.py

```diff
@@ -0,0 +1,590 @@
+                f.write(rel.stderr)
+
+            run_instrumented(True, fn, targetdir)
+            run_instrumented(False, fn, targetdir)
+
+
+def print_cols(dbg, release, additional=""):
+    print(
+        (dbg.ljust(OUTPUT_COLUMN_WIDTH))
+        + release.ljust(OUTPUT_COLUMN_WIDTH)
+        + additional
+    )
+
+
+def compare_cflog(dbg_tdfile, rel_tdfile):
+    import json
+
+    def get_cflog_entires(tdfile, is_debug):
+        with open(function_id_path(is_debug)) as f:
+            function_id = json.load(f)
+        cflog = tdfile._get_section(taint_dag.TDControlFlowLogSection)
+        cflog.function_id_mapping(list(map(cxxfilt.demangle, function_id)))
+        return list(
+            map(
+                lambda e: (input_offsets(e.label, tdfile), e.callstack),
+                filter(
+                    lambda e: isinstance(e, taint_dag.TDTaintedControlFlowEvent), cflog
+                ),
+            )
+        )
+
+    dbg = get_cflog_entires(dbg_tdfile, True)
+    rel = get_cflog_entires(rel_tdfile, False)
+
+    print("COMPARE CONTROL FLOW LOGS")
+    n = max(len(dbg), len(rel))
+
+    len_dbg = len(dbg)
+    len_rel = len(rel)
+
+    dbgidx = 0
+    relidx = 0
+    while dbgidx < len_dbg or relidx < len_rel:
+        dbg_entry = dbg[dbgidx] if dbgidx < len_dbg else None
+        rel_entry = rel[relidx] if relidx < len_rel else None
+
+        if dbg_entry is None:
+            print_cols("", str(rel_entry), "")  # , rel_funcname(rel_entry[1]))
+            relidx += 1
+            continue
+
+        if rel_entry is None:
+            print_cols(str(dbg_entry), "", "")  # dbg_funcname(dbg_entry[1]))
+            dbgidx += 1
+            continue
+
+        dbg_callstack = str(dbg_entry[1])
+        rel_callstack = str(rel_entry[1])
+
+        if dbg_entry[0] == rel_entry[0]:
+            print_cols(
+                str(dbg_entry[0]),
+                str(rel_entry[0]),
+                f" !!! DBG: {dbg_callstack} != REL: {rel_callstack}"
+                if dbg_callstack != rel_callstack
+                else "",
+            )
+            dbgidx += 1
+            relidx += 1
+        else:
+            # check if we should be stepping debug or release
+            # depending on shortest path
+            debug_steps = 0
+            release_steps = 0
+
+            while dbgidx + debug_steps < len_dbg:
+                if dbg[dbgidx + debug_steps][0] == rel_entry[0]:
+                    break
+                debug_steps += 1
+
+            while relidx + release_steps < len_rel:
+                if rel[relidx + release_steps][0] == dbg_entry[0]:
+                    break
+                release_steps += 1
+
+            if debug_steps < release_steps:
+                print_cols(str(dbg_entry[0]), "", dbg_callstack)
+                dbgidx += 1
+            else:
+                print_cols("", str(rel_entry[0]), rel_callstack)
+                relidx += 1
+
+    return
+
+    # dbg_offset_to_func = {}
```

Comment:
There is some commented out code here and there. I'd recommend we just remove it.

Reply:
Can do